pageserver/src/page_service.rs

```diff
@@ -607,8 +577,22 @@ impl PageServerHandler {
         // Import wal provided via CopyData
         info!("importing wal");
         pgb.write_message(&BeMessage::CopyInResponse)?;
-        let reader = CopyInReader::new(pgb);
-        import_wal_from_tar(&*timeline, reader, start_lsn, end_lsn)?;
+        pgb.flush().await?;
+        let mut copyin_stream = Box::pin(copyin_stream(pgb));
+        let reader = SyncIoBridge::new(StreamReader::new(&mut copyin_stream));
+        tokio::task::block_in_place(|| {
```

Comment:
I wonder, why use `block_in_place` instead of `spawn_blocking`? 
I would imagine, that it would be better to explicitly spawn tasks for blocking operations, instead of implicitly forcing other tasks to move away to other worker threads.